src/network/networkd-bridge-mdb.c

```diff
@@ -252,30 +263,46 @@ static int bridge_mdb_verify(BridgeMDB *mdb) {
-
-        if (mdb->family == AF_INET) {
-                if (in4_addr_is_local_multicast(&mdb->group_addr.in))
+        switch (mdb->type) {
+        case BRIDGE_MDB_ENTRY_TYPE_L2:
+                if (!ether_addr_is_multicast(&mdb->l2_addr))
+                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "%s: MulticastGroupAddress= is not an L2 multicast address. "
+                                "Ignoring [BridgeMDB] section from line %u.",
+                                mdb->section->filename, mdb->section->line);
+                break;
+        case BRIDGE_MDB_ENTRY_TYPE_L3:
+                if (mdb->family == AF_UNSPEC)
                         return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                 "%s: MulticastGroupAddress= is a local multicast address. "
+                                                 "%s: [BridgeMDB] section without MulticastGroupAddress= field configured. "
                                                  "Ignoring [BridgeMDB] section from line %u.",
                                                  mdb->section->filename, mdb->section->line);
-        } else {
-                if (in6_addr_is_link_local_all_nodes(&mdb->group_addr.in6))
+
+                if (!in_addr_is_multicast(mdb->family, &mdb->group_addr))
                         return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                 "%s: MulticastGroupAddress= is the multicast all nodes address. "
+                                                 "%s: MulticastGroupAddress= is not a multicast address. "
                                                  "Ignoring [BridgeMDB] section from line %u.",
                                                  mdb->section->filename, mdb->section->line);
+
+                if (mdb->family == AF_INET) {
+                        if (in4_addr_is_local_multicast(&mdb->group_addr.in))
+                                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                         "%s: MulticastGroupAddress= is a local multicast address. "
+                                                         "Ignoring [BridgeMDB] section from line %u.",
+                                                         mdb->section->filename, mdb->section->line);
+                } else {
+                        if (in6_addr_is_link_local_all_nodes(&mdb->group_addr.in6))
+                                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                         "%s: MulticastGroupAddress= is the multicast all nodes address. "
+                                                         "Ignoring [BridgeMDB] section from line %u.",
+                                                         mdb->section->filename, mdb->section->line);
+                }
```

Comment:
coding style: no {} for single line if blocks

Reply:
This is no longer relevant, because the if/else was replaced by a switch statement.